pkg/auth0-cli-config-generator/main.go

```diff
@@ -0,0 +1,206 @@
+// auth0-cli-config-generator: A command that generates a valid config file that can be used with auth0-cli.
+//
+// Currently this command is only used to generator a config using environment variables which is then used for integration tests.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type params struct {
+	filePath     string
+	clientName   string
+	clientDomain string
+	clientID     string
+	clientSecret string
+}
+
+func (p params) validate() error {
+	if p.clientName == "" {
+		return fmt.Errorf("Missing client name")
+	}
+
+	if p.clientDomain == "" {
+		return fmt.Errorf("Missing client domain")
+	}
+
+	u, err := url.Parse(p.clientDomain)
+	if err != nil {
+		return fmt.Errorf("Failed to parse client domain: %s", p.clientDomain)
+	}
+
+	if u.Scheme != "" {
+		return fmt.Errorf("Client domain cant include a scheme: %s", p.clientDomain)
+	}
+
+	if p.clientID == "" {
+		return fmt.Errorf("Missing client id")
+	}
+
+	if p.clientSecret == "" {
+		return fmt.Errorf("Missing client secret")
+	}
+	return nil
+}
+
+type config struct {
+	DefaultTenant string            `json:"default_tenant"`
+	Tenants       map[string]tenant `json:"tenants"`
+}
+
+type tenant struct {
+	Name        string    `json:"name"`
+	Domain      string    `json:"domain"`
+	AccessToken string    `json:"access_token,omitempty"`
+	ExpiresAt   time.Time `json:"expires_at"`
+}
+
+func isLoggedIn(filePath string) bool {
+	var c config
+	var buf []byte
+	var err error
+	if buf, err = ioutil.ReadFile(filePath); err != nil {
```

Comment:
Go is big on the backward-compability thing, and I'm glad for it. :)

That said, in 1.16 they are suggesting people switch to `os.ReadFile`

https://www.srcbeat.com/2021/01/golang-ioutil-deprecated/

Reply:
Thanks 🙇 I've replaced `ioutil.ReadFile` and `ioutil.WriteFile` with `os.ReadFile` and `os.WriteFile`.